nassl/openssl3_ssl_client.py

```diff
@@ -0,0 +1,177 @@
+"""SSL client implementation using OpenSSL 3.x."""
```

Comment:
given the experimental project status, maybe it makes since to move this into modern ssl and move the current modern ssl to legacy (but this would remove support for the existing legacy client)